apps/x509.c

```diff
@@ -1064,8 +1062,6 @@ static int sign(X509 *x, EVP_PKEY *pkey, int days, int clrext,
         goto err;
     if (!preserve_dates && !set_cert_times(x, NULL, NULL, days))
         goto err;
-    if (!X509_set_pubkey(x, pkey))
-        goto err;
```

Comment:
This change looks like it will change the semantics of the "-signkey" option. From the docs for "-signkey":

````
If the input file is a certificate it sets the issuer name to the
subject name (i.e.  makes it self signed) changes the public key to the
supplied value and changes the start and end dates.
````

By removing this line it looks to me like "-signkey" would no longer "change the public key to the supplied value".

Reply:
Following the flow of `x` through `x509_main()` in case `reqfile == 0` (i.e., `-req` is not given), it is read from the input (either the file provided with `-in` or stdin), such that its default public key is the one contained in the input cert, and that this is optionally overridden by the argument of `-force_pubkey`, which was read into `fkey`:
```
        x = load_cert(infile, informat, "Certificate");
        if (x == NULL)
            goto end;
        if (fkey != NULL && !X509_set_pubkey(x, fkey))
            goto end;
```
Subsequently, until
```
    if (!sign(x, Upkey, days, clrext, digest, extconf, extsect, preserve_dates))
```
its public key is not changed.
If in the called function the code I removed:
```
    if (!X509_set_pubkey(x, pkey))
       goto err;
```
was still present, the supplied pubkey value would be overridden by the pubkey part of the signing key `pkey`, which would defeat the purpose of `-force_pubkey`. On the other hand, I now see that without `-force_pubkey` the public key related to `pkey` should be set.


To address your point, I think two things are due:
* update the doc by adding after "changes the public key to the supplied value" the restriction "unless overridden by -force_pubkey"
* restore `X509_set_pubkey(x, pkey)` in case `-force_pubkey` is not present.

I've just done this. Ok now?

Doing so, I found that the doc for `-signkey` in the `-req` case:

> If the input is a certificate request then a self-signed certificate
 is created using the supplied private key using the subject name in
 the request.

did not address the `-day` and `-clrext` option, which do have effect there as well.
Since this case effectively gets handled quite the same as the cert input case I combined the two.

I also streamlined the wording in x509.pod a little and improved the layout of one paragraph.